class.jetpack-settings.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/**
+ * All the Settings we want to sync to .com
+ *
+ * This is also used when displaying settings in the API
+ */
+class Jetpack_Settings {
+
+	public static $settings = array(
+		// also exists as "options"
+		'default_ping_status'     => 'bool',
+		'default_comment_status'  => 'bool',
+
+		// new stuff starts here
+		'blog_public'             => 'int',
+		'jetpack_sync_non_public_post_stati' => 'bool',
+		'jetpack_relatedposts_allowed' => 'bool',
+		'jetpack_relatedposts_enabled' => 'bool',
+		'jetpack_relatedposts_show_headline' => 'bool',
+		'jetpack_relatedposts_show_thumbnails' => 'bool',
+		'default_category'        => array( 'ID' =>'int', 'name' => 'string' ),
+		'post_categories'         => array( 'ID' =>'int', 'name' => 'string' ),
+		'default_post_format'     => 'string', // to check
+		'default_pingback_flag'   => 'bool',
+		'require_name_email'      => 'bool',
+		'comment_registration'    => 'bool',
+		'close_comments_for_old_posts' => 'bool' ,
+		'close_comments_days_old' => 'int' ,
+		'thread_comments'         => 'bool' ,
+		'thread_comments_depth'   => 'int' ,
+		'page_comments'           => 'bool' ,
+		'comments_per_page'       => 'int' ,
+		'default_comments_page'   => 'string' , // to check
+		'comment_order'           => 'string', //  to check
+		'comments_notify'         => 'bool' ,
+		'moderation_notify'       => 'bool',
+		'social_notifications_like' => 'bool',
+		'social_notifications_reblog' => 'bool',
+		'social_notifications_subscribe' => 'bool',
+		'comment_moderation'      => 'bool',
+		'comment_whitelist'       => 'bool',
+		'comment_max_links'       => 'int',
+		'moderation_keys'         => 'string', // to check
+		'blacklist_keys'          => 'bool', // to check
+		'lang_id'                 => 'bool', // to check
+		'wga'                     => 'bool', // array to check
+		'disabled_likes'          => 'bool',
+		'disabled_reblogs'        => 'bool',
+		'jetpack_comment_likes_enabled' => 'bool',
+		'twitter_via'             => 'string',
+		'jetpack-twitter-cards-site-tag' => 'string'
+
+	);
+
+	/**
+	 * Data that we want to sync.
+	 * @var array
+	 */
+	public static $mock_options = array(
+		'admin_url' => array( 'type' => 'url', 'callback' => 'get_admin_url', 'callback_' )
```

Comment:
The last element of the array here seems wrong. Should that be `'callback_args' => ''` maybe?